### ISSUE TITLE
BUG: Authentication in SDK 1.1.5.1

### ISSUE THREAD

**Issue by swanderz** [Azure/MachineLearningNotebooks#871]
@thwagne tells me this is already a known issue. 
here's a [reproducible example of the bug](https://github.com/swanderz/MachineLearningNotebooks/blob/auth_error/how-to-use-azureml/machine-learning-pipelines/intro-to-pipelines/aml-pipelines-with-data-dependency-steps.ipynb) using the Azure Machine Learning Pipelines with Data Dependency notebook
cc: @rastala @lostmygithubaccount @sanpil 
```python
Traceback (most recent call last):
  File "model_drift.py", line 37, in model_drift
    for run in exp.get_runs():
  File "/azureml-envs/azureml_0cd9634699e0e6aba087adfc14a9234b/lib/python3.6/site-packages/azureml/core/run.py", line 390, in _rehydrate_runs
    yield factory(experiment, run_dto)
  File "/azureml-envs/azureml_0cd9634699e0e6aba087adfc14a9234b/lib/python3.6/site-packages/azureml/pipeline/core/run.py", line 386, in _from_dto
    return PipelineRun(experiment=experiment, run_id=run_dto.run_id)
  File "/azureml-envs/azureml_0cd9634699e0e6aba087adfc14a9234b/lib/python3.6/site-packages/azureml/pipeline/core/run.py", line 80, in __init__
    service_endpoint=_service_endpoint)
  File "/azureml-envs/azureml_0cd9634699e0e6aba087adfc14a9234b/lib/python3.6/site-packages/azureml/pipeline/core/_graph_context.py", line 46, in __init__
    service_endpoint=service_endpoint)
  File "/azureml-envs/azureml_0cd9634699e0e6aba087adfc14a9234b/lib/python3.6/site-packages/azureml/pipeline/core/_aeva_provider.py", line 127, in create_provider
    service_endpoint=service_endpoint)
  File "/azureml-envs/azureml_0cd9634699e0e6aba087adfc14a9234b/lib/python3.6/site-packages/azureml/pipeline/core/_aeva_provider.py", line 142, in create_service_caller
    service_endpoint = _AevaWorkflowProvider.get_endpoint_url(workspace, experiment_name)
  File "/azureml-envs/azureml_0cd9634699e0e6aba087adfc14a9234b/lib/python3.6/site-packages/azureml/pipeline/core/_aeva_provider.py", line 163, in get_endpoint_url
    workspace_discovery_url=workspace.discovery_url)
  File "/azureml-envs/azureml_0cd9634699e0e6aba087adfc14a9234b/lib/python3.6/site-packages/azureml/core/workspace.py", line 719, in discovery_url
    self.get_details()
  File "/azureml-envs/azureml_0cd9634699e0e6aba087adfc14a9234b/lib/python3.6/site-packages/azureml/core/workspace.py", line 779, in get_details
    self._subscription_id)
  File "/azureml-envs/azureml_0cd9634699e0e6aba087adfc14a9234b/lib/python3.6/site-packages/azureml/_project/_commands.py", line 526, in show_workspace
    AzureMachineLearningWorkspaces, subscription_id).workspaces,
  File "/azureml-envs/azureml_0cd9634699e0e6aba087adfc14a9234b/lib/python3.6/site-packages/azureml/core/authentication.py", line 144, in _get_service_client
    all_subscription_list, tenant_id = self._get_all_subscription_ids()
TypeError: 'NoneType' object is not iterable
```

**Comment by ram-msft** [comment]
@swanderz Thanks for the feedback. We are investigating the issue and will update you shortly.

**Comment by shbijlan** [comment]
Thank you for your feedback. I have opened a work item to investigate this. We will update you as we have more information.

**Comment by shbijlan** [comment]
@swanderz this issue was fixed in azureml-core 1.1.5.5.
